### ISSUE TITLE
[CI] ClusterSearchShardsResponseTests testSerialization failing

### ISSUE THREAD

**Issue by ldematte** [elastic/elasticsearch#100289]
After version upgrade to 8.12, 8.11 can be generated as a random version to be passed to DiscoveryNodeUtils.create (by `VersionUtils.randomVersion(random())`). However from 8.10 forward, minIndexVersion, maxIndexVersion need to be specified too. 
The test should be updated to use DiscoveryNode.builder() directly and specify min/max index versions too.

**Build scan:**
https://gradle-enterprise.elastic.co/s/ib3thvzxhoyo4/tests/:server:test/org.elasticsearch.action.admin.cluster.shards.ClusterSearchShardsResponseTests/testSerialization
**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.action.admin.cluster.shards.ClusterSearchShardsResponseTests.testSerialization" -Dtests.seed=194A3CC5493A6758 -Dtests.locale=et -Dtests.timezone=GMT0 -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.action.admin.cluster.shards.ClusterSearchShardsResponseTests&tests.test=testSerialization
**Failure excerpt:**
```
java.lang.IllegalArgumentException: Node versions can only be inferred before release version 8.10.0

  at __randomizedtesting.SeedInfo.seed([194A3CC5493A6758:4D925D8D6E685AC4]:0)
  at org.elasticsearch.cluster.node.VersionInformation.inferVersions(VersionInformation.java:42)
  at org.elasticsearch.cluster.node.DiscoveryNodeUtils$Builder.build(DiscoveryNodeUtils.java:144)
  at org.elasticsearch.cluster.node.DiscoveryNodeUtils.create(DiscoveryNodeUtils.java:37)
  at org.elasticsearch.action.admin.cluster.shards.ClusterSearchShardsResponseTests.testSerialization(ClusterSearchShardsResponseTests.java:52)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

**Comment by ldematte** [comment]
I see this failing again, with the same exception: https://gradle-enterprise.elastic.co/s/ny6d4ym32nwvy/tests/task/:server:test/details/org.elasticsearch.action.admin.cluster.shards.ClusterSearchShardsResponseTests/testSerialization?top-execution=1

https://gradle-enterprise.elastic.co/s/cltby2i2mi22m/tests/task/:server:test/details/org.elasticsearch.action.admin.cluster.shards.ClusterSearchShardsResponseTests/testSerialization?top-execution=1

Not sure why: maybe related to 8.11 release?

**Comment by benwtrent** [comment]
🤔 weird that it failed again. I will look

**Comment by mark-vieira** [comment]
Yep, this sounds like fallout from 8.11.0. This error is happening across a number of tests. This is likely a core/infra issue. I recall a similar set of errors when 8.10 released.

![image](https://github.com/elastic/elasticsearch/assets/4106672/0d38f68b-f6d1-48ee-9a06-658c75de49a2)

**Comment by ldematte** [comment]
> weird that it failed again. I will look

Yes, the weird thing is that the fix you merged should have fixed it permanently. Maybe it's failing elsewhere - I will also give it a look.

**Comment by ldematte** [comment]
Ah, I see. The test was fixed after 8.11 FF, and now after 8.11 released we are doing 8.11 branch CI - which shows the same error again. I will backport the fix to the 8.11 branch
